src/components/media/MediasSelectModal.jsx

```diff
@@ -127,38 +159,103 @@ const MediasSelectModal = ({
     searchValue
   ).map(({ name }) => name)
 
-  const { watch, handleSubmit } = useFormContext()
-  return (
-    <Modal
-      isOpen
-      onClose={onClose}
-      size="6xl"
-      scrollBehavior="inside"
-      closeOnOverlayClick={false}
-      isCentered
-    >
-      <ModalOverlay />
-      <ModalContent padding="0.5rem" paddingTop="1rem" maxHeight="90%">
-        <ModalHeader>
-          <VStack alignItems="right" gap="1rem">
-            <HStack justifyContent="space-between">
-              <Text textStyle="h4">{`Add ${mediaRoom.slice(0, -1)}${
-                fileName ? ` to ${decodeURIComponent(fileName)}` : ""
-              }`}</Text>
-              <ModalCloseButton onClick={onClose} position="static" />
-            </HStack>
-            <Text textStyle="body-1">
-              Choose from your {pluralMediaLabel} or upload a new{" "}
-              {singularMediaLabel}. You can organise your {pluralMediaLabel} in
-              Workspace &gt; {titleCase(pluralMediaLabel)}.
-            </Text>
-            <Flex w="100%" justifyContent="space-between">
+  const [activeTab, setActiveTab] = useState(0)
+
+  const handleTabChange = (index) => {
+    onMediaSelect("")
+    clearErrors()
+    setActiveTab(index)
+  }
+
+  const debouncedHandleSearch = _.debounce((value) => {
+    setSearchedValue(value)
+    // Perform any other actions you want with the input value
+  }, 1000)
+
+  const internalMediaSelect = (
+    <Grid h="100%" templateColumns="15.25rem 1fr">
+      <GridItem borderColor="base.divider.medium" overflowY="auto" h="50vh">
+        <VStack pr="1.25rem">
+          {queryParams.mediaDirectoryName &&
+            queryParams.mediaDirectoryName !== mediaType && (
+              <Button
+                w="100%"
+                justifyContent="start"
+                variant="clear"
+                alignSelf="start"
+                leftIcon={<Icon as={BiLeftArrowAlt} fontSize="1.25rem" />}
+                color="base.content.default"
+                onClick={() => {
+                  onMediaSelect("")
+                  setQueryParams((prevState) => {
+                    return {
+                      ...prevState,
+                      mediaDirectoryName: getMediaDirectoryName(
+                        queryParams.mediaDirectoryName,
+                        {
+                          end:
+                            queryParams.mediaDirectoryName.split("%2F").length -
+                            1,
+                        }
```

Comment:
could i get some clarity on what this is doing? specifically, what the `end` parameter is doing here? 

i'm abit confused as to why we wouldn't just use the default `mediaDirectoryName` and have to add an additional `end` parameter